src/api/spec/controllers/webui/project_controller_spec.rb

```diff
@@ -369,6 +369,76 @@
     end
   end
 
+  describe 'POST #save_distribution' do
+    let(:action) { post :save_distributions, project: apache_project, distributions: ['openSUSE_Tumbleweed'] }
+
+    before do
+      login admin_user
+      Project.any_instance.stubs(:save_distributions).returns(true)
+      Project.any_instance.stubs(:prepend_kiwi_config).returns()
+      Project.any_instance.stubs(:store).returns(true)
+    end
+
+    it "should call 'add_distributions' with [ 'openSUSE_Tumbleweed' ]" do
+      Project.any_instance.expects(:save_distributions).with(['openSUSE_Tumbleweed']).returns(true).once
+      action
+    end
+
+    context 'with valid distribution attributes' do
+      before do
+        action
+      end
+
+      it { is_expected.to redirect_to(project_repositories_path(apache_project)) }
+      it { expect(flash[:success]).to eq('Successfully added repositories') }
+    end
+
+    context 'with invalid distribution attributes' do
+      before do
+        # http://stackoverflow.com/questions/6040479/rspec-testing-redirect-to-back
+        request.env['HTTP_REFERER'] = 'where_i_came_from'
+        # It's necessary to call apache_project, otherwise create will fail due to the error stub
+        apache_project
+        Project.any_instance.stubs(:distribution_errors).returns('error')
+        Project.any_instance.stubs(:save_distributions).returns(false)
+        action
+      end
+
+      it { is_expected.to redirect_to('where_i_came_from') }
+      it { expect(flash[:error]).to eq("Can't add repositories: error") }
+    end
+
+    context 'with valid image attribute' do
+      let(:action) { post :save_distributions, project: apache_project, images: true }
+
+      it { expect(action).to redirect_to(project_repositories_path(apache_project)) }
+
+      it 'should eq Successfully added repositories' do
+        action
+        expect(flash[:success]).to eq('Successfully added repositories')
+      end
+
+      it "should call 'prepend_kiwi_config'" do
+        Project.any_instance.expects(:prepend_kiwi_config).once
+        action
+      end
+    end
+
+    context 'with valid distribution and image attribute' do
+      let(:action) { post :save_distributions, project: apache_project, images: true, distributions: ['openSUSE_Tumbleweed'] }
+
+      it "should call 'save_distributions' with [ 'openSUSE_Tumbleweed' ]" do
+        Project.any_instance.expects(:save_distributions).with(['openSUSE_Tumbleweed']).once
+        action
+      end
+
+      it "should call 'prepend_kiwi_config'" do
+        Project.any_instance.expects(:prepend_kiwi_config).once
+        action
+      end
+    end
+  end
```

Comment:
Hm! But not in combination! I can remove it if you think it does not add value!
(but maybe you're right because the two cases are somehow independent)